### PR TITLE
Added a workflow for automatic release uploads

### PR DESCRIPTION
**Pull request by parwam**
- Updated github workflow to trigger build checks on version tag push and to automatically upload builds as releases on manual trigger from publish releases
- Updated CMakeLists.txt to collect source and header files from subfolders as well
- Updated CMakeLists.txt to copy resources folder to binary output directory to handle resource dependency for the executable
- Updated README.md (fixed some typos)

### CODE REVIEW

File: .github/workflows/build-release.yml [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+          path: ${{ github.workspace }}/build/sfml-project-macos.zip
+
+      - name: Upload Linux Build
+        if: matrix.os == 'ubuntu-latest'
+        uses: actions/upload-artifact@v3
+        with:
+          name: sfml-project-linux
+          path: ${{ github.workspace }}/build/sfml-project-linux.zip
+
+      release:
```
Comment (ufrshubham) [inline .github/workflows/build-release.yml]: Just a minor formatting suggestion
```suggestion
  release:
```

**Review by ufrshubham — CHANGES_REQUESTED** [review]
LGTM! Thanks for the PR. I've just added a minor suggestion🙂

**Comment by parwam** [comment]
Fixed it.

**Review by ufrshubham — APPROVED** [review]

**Comment by ufrshubham** [comment]
@MockJoke is it possible for you to also update the readme with steps on how to publish a release?

**Pull request merged**